src/analyze/analyze-capability.c

```diff
@@ -20,27 +33,40 @@ int verb_capabilities(int argc, char *argv[], void *userdata) {
         /* Determine the maximum of the last cap known by the kernel and by us */
         last_cap = MAX((unsigned) CAP_LAST_CAP, cap_last_cap());
 
-        if (strv_isempty(strv_skip(argv, 1)))
+        if (argc == 1)
                 for (unsigned c = 0; c <= last_cap; c++) {
-                        r = table_add_many(table,
-                                           TABLE_STRING, capability_to_name(c) ?: "cap_???",
-                                           TABLE_UINT, c);
+                        r = table_add_capability(table, c);
                         if (r < 0)
-                                return table_log_add_error(r);
+                                return r;
                 }
-        else {
+        else if (argc == 2 && safe_atoux64(argv[1], &cap_mask) >= 0) {
```

Comment:
hmm, this creates an ambiguity, because capability_from_name() actually takes numeric caps names too. i.e. "3" could mean CAP_FOWNER, or it could mean CAP_DAC_OVERRIDE|CAP_DAC_READ_SEARCH, it's not clear from the context.

Hence, either add a switch `--mask`/`-m` that selects this mode, or alternatively maybe check for some prefix like `@` or so that indicates the mask.

i guess the switch thing is less surprising.

Reply:
yeah, sorry, apparently i skipped over the PR message. i am seeing that now too

Reply:
yes, make it global.